### PR TITLE
alias update, clean get response

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):
- refactored `get` response object and removed unnecessary data
- generated method for handling updating personal information (contact, address, aliases), prevent history record in partial save mode
- refactored manager test
- endpoint for update application
- cleanup, mappings and validation

### CODE REVIEW

File: src/Spd.Manager.Licence/ControllingMemberCrcAppContract.cs [hunk 1]
```diff
@@ -42,30 +43,49 @@ public record ControllingMemberCrcAppBase
...
+    public Guid BizContactId { get; set; }
+    public Guid InviteId { get; set; }
+    public Guid? ParentBizLicApplicationId { get; set; }
 };
 #endregion
 #region anonymous user
 public record ControllingMemberCrcAppSubmitRequest : ControllingMemberCrcAppBase
 {
+    public Guid? ControllingMemberAppId { get; set; }
+    public Guid? ApplicantId { get; set; }
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppContract.cs:71]: I think we talked about this. the applicantId should not be here.

**Review by peggy-quartech — CHANGES_REQUESTED** [review]

File: src/Spd.Manager.Licence/ControllingMemberCrcAppContract.cs (hunk 1 above)
Comment (esdd1995) [reply src/Spd.Manager.Licence/ControllingMemberCrcAppContract.cs:71]: fe doesn't need to fill that, I will remove it from request object later

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 2]
```diff
@@ -125,4 +123,50 @@ public async Task<ControllingMemberCrcApplicationCmdResp> SaveControllingMemberC
         return new ControllingMemberCrcApplicationCmdResp((Guid)app.spd_applicationid, cmd.ContactId);
     }
 
+    //update contact, aliases and addreses, support partial save (to not add spd_address and spd_alias histroy record)
+    private async Task<contact?> UpdatePersonalInformationAsync(SaveControllingMemberCrcAppCmd cmd, contact? contact, CancellationToken ct)
+    {
+        List<AliasResp> aliases = _mapper.Map<List<AliasResp>>(contact?.spd_Contact_Alias.Where(a=>a.statecode == DynamicsConstants.StateCode_Active &&
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:129]: why we need to map to aliasResp?

**Review by peggy-quartech — APPROVED** [review]

File: src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs [hunk 3]
```diff
@@ -125,4 +123,50 @@ public async Task<ControllingMemberCrcApplicationCmdResp> SaveControllingMemberC
...
+                a.spd_source == (int)AliasSourceTypeOptionSet.UserEntered));
+        contact newContact = _mapper.Map<contact>(cmd);
+        List<spd_alias> aliasesToAdd = (List<spd_alias>)_mapper.Map<IEnumerable<spd_alias>>(cmd.Aliases.Where(a => a.Id == null || a.Id == Guid.Empty)); // Only aliases with Id null or empty are considered as new
+        var modifiedAliases = cmd.Aliases.Where(a => a.Id != Guid.Empty && a.Id != null).ToList();
+        List<Guid?> aliasesToRemove = aliases.Where(a => modifiedAliases.All(ap => ap.Id != a.Id)).Select(a => a.Id).ToList();
+
+        contact = await _context.UpdateContact(contact, newContact, null, aliasesToAdd, ct, cmd.IsPartialSaving);
+        foreach (var aliasId in aliasesToRemove)
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs:137]: I would like to see alias remove also merged into UpdateContact

**Pull request merged**